ui/src/publication/publication-model.ts

```diff
@@ -37,12 +37,30 @@ export enum DraftChangeType {
 
 export type Operation = 'CREATE' | 'DELETE' | 'MODIFY' | 'RESTORE' | 'CALCULATED';
 
+export type PublicationGroupId = string;
+export type PublicationGroup = {
+    id: PublicationGroupId;
+};
+
+export enum PublicationStage {
+    UNSTAGED = 'UNSTAGED',
+    STAGED = 'STAGED',
+}
```

Comment:
Ei suoraan tämän tiketin asiaa eikä tätä tarvi muuttaa jollet halua, mutta meillä on käytetty jonkin verran ristiin näitä "oikeita" TS:n enumeja ja ihan vaan string-vakioita. Tämänhän siis voisi kirjoittaa myös `export type PublicationStage = 'UNSTAGED' | 'STAGED'`. Voitaisiin ehkä jossain kohtaa miettiä näiden käytäntöjen yhdistämistä.

Itsetykkään enemmän tuosta string-vakiotavasta, sillä TypeScriptin enumien luonteen takia niiden asioiden tekeminen mitä enumeilla yleisesti yritetään saavuttaa (mäppäys arvoihin, järjestäminen jne.) ei mene yhtä kohtuuella kuin muissa kielissä (plus bäkkärihän palauttaa stringejä.) Mutta lopulta mulle ei ole niin väliä kumpaan malliin päädytään, lähinnä halusin nostaa esille että yhtenäistämisen tarvetta voisi olla.

Reply:
Näissäkin on vähän puolensa. Mä suosin itse noita varsinaisia enumeja, koska silloin nimitykseen liittyvä konteksti pysyy mukana ja käyttötapa on aina samanlainen vs. stringeihin. Tästähän jotain jutskailtiin aiemmin siinä DRAFT/OFFICIAL-direct importin kanssa.

Jos siis jossakin näkee pelkästään `'STAGED'`/`'UNSTAGED'`, niin voi olla hankalampaa hahmottaa mistä on kyse, verrattuna kontekstia kuvaavaan `PublicationStage.UNSTAGED`/`PublicationStage.STAGED`. (Tässä yhteydessä stagetustermi ei ole toistaiseksi muualla käytössä, joten hähmeysriski on pienempi).

Jos myös arvo on jo valmiiksi järkevä, niin riski on myös pienempi (tässä toteutuu). Jos arvoksi tarvittaisiin esim optimointitarkoituksessa lyhenne "u" tai "s", niin silloin enumin käyttäminen olisi hyödyllisempää eksplisiittisyyden vuoksi.

Käännöserona tietysti myös, että enumit päätyvät myös suoritusaikaiseksi asioiksi, kun taas string-unioni ei. Enumi on siis mahdollista esim. iteroida läpi: `const asd = Object.keys/.values/.entries(PublicationStage)`

Reply:
Olikos muuten Jyrkillä jokin suuntakuvio näihin? Jätin toistaiseksi enumin olemassaolevaksi tähän.